src/main/java/uk/gov/register/views/PreviewPageView.java

```diff
@@ -0,0 +1,117 @@
+
+import javax.inject.Provider;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+
+public abstract class PreviewPageView extends ThymeleafView {
+
+    private static final String NUMBER_FORMAT = "#,###";
+    private static final String PREVIEW_HTML_PAGE = "preview.html";
+
+    protected static final String YAML = "YAML";
+    protected static final String TTL = "TTL";
+    protected static final String JSON = "JSON";
+
+    private final String key;
+    private final String previewType;
+    private final HomepageContent homepageContent;
+    private final Provider<RegisterName> registerNameProvider;
+    private final ElementType elementType;
+    private final int totalRecords;
+    private final int totalEntries;
+
+    public PreviewPageView(final RequestContext requestContext,
+                           final RegisterTrackingConfiguration registerTrackingConfiguration,
+                           final RegisterResolver registerResolver,
+                           final RegisterReadOnly register,
+                           final String key,
+                           final String previewType,
+                           final ElementType elementType,
+                           final HomepageContent homepageContent,
+                           final Provider<RegisterName> registerNameProvider) {
+        super(requestContext, PREVIEW_HTML_PAGE, registerTrackingConfiguration, registerResolver, register);
+        this.key = key;
+        this.previewType = previewType;
+        this.elementType = elementType;
+        this.homepageContent = homepageContent;
+        this.registerNameProvider = registerNameProvider;
+        totalRecords = register.getTotalRecords();
+        totalEntries = register.getTotalEntries();
+    }
+
+    public abstract String getRegisterValues();
+
+    @SuppressWarnings("unused, used by templates")
+    public String getTotalElements() {
+        final NumberFormat nf = new DecimalFormat(NUMBER_FORMAT);
+
+        return nf.format(getTotalObjects());
+    }
+
+    public String getPreviewTypeTag() {
+        if (ExtraMediaType.TEXT_YAML_TYPE.getSubtype().equals(previewType)) {
+            return YAML;
+        } else if (ExtraMediaType.TEXT_TTL_TYPE.getSubtype().equals(previewType)) {
+            return TTL;
+        } else {
+            return JSON;
+        }
+    }
+
+    @SuppressWarnings("unused, used by templates")
+    public String getUrlMultipleElements() {
+        if (elementType.getMultipleElementsDownloadLocation().isEmpty()) {
+            return StringUtils.EMPTY;
+        }
+
+        return String.format(elementType.getMultipleElementsDownloadLocation(), getPreviewTypeTag().toLowerCase());
+    }
+
+    @SuppressWarnings("unused, used by templates")
+    public String getUrlSingleElements() {
+        return String.format(elementType.getSingleElementDownloadLocation(), getKey(), getPreviewTypeTag().toLowerCase());
+    }
+
+    public abstract int getTotalObjects();
```

Comment:
What is total objects?

Reply:
A generic method to recover the number of objects available in the view, for records return number of records, for entries, number of entries and for the item 1

Reply:
Discussed.